### PR TITLE
feat: new page for windows management

### PR DESCRIPTION
**Pull request by ElviraMustafina**
- separate page for top-level windows management
- new gifs
- minor updates to code samples like titles for windows

note: the topic is temporarily placed top-level inside "Compose Multiplatform UI framework", I'll move it under "Desktop-specific components" when other parts are ready

### CODE REVIEW

**Review by m-sasha — APPROVED** [review]

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 1]
```diff
@@ -0,0 +1,705 @@
...
+    }
+}
+```
+{initial-collapse-state="collapsed" collapsed-title="Window(onCloseRequest = { isVisible = false },"}
+
+<img src="compose-window-hide-tray.png" alt="Hide instead of closing" animated="true" width="600"/>
+
+## Function `singleWindowApplication`
+
+`singleWindowApplication` is a function for the simplified creation of a single window application:
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:274]: To get rid of nominalizations as much as possible, I would rephrase here to something like "You can create a single window application by calling the `singleWindowApplication` function". And if the "easier to use" paragraph has been moved above, we don't need to mention that it is easier again.

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 2]
```diff
@@ -0,0 +1,705 @@
...
+```
+{initial-collapse-state="collapsed" collapsed-title="state = rememberWindowState(width = Dp.Unspecified, height = Dp.Unspecified)"}
+
+<img src="compose-window-adaptive-size.png" alt="Adaptive window size" width="451"/>
+
+## Changing the window state
+
+`WindowState` is a separate API class for the window placement, current position, and size. The placement attribute allows you to specify how the window is placed on the screen: 
+floating, maximized/minimized, or fullscreen.
+To change the window state, use callbacks or observe it in composables. Any change of the state triggers automatic recomposition.
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:337]: I think the order here would be more natural with "Any change..." first and "To change the window state..." second. And then a colon after the second sentence to transition into the code block.

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 3]
```diff
@@ -0,0 +1,705 @@
...
+    }
+}
+```
+{initial-collapse-state="collapsed" collapsed-title="val state = rememberWindowState(placement = WindowPlacement.Maximized)"}
+
+<img src="compose-window-minimize.png" alt="Changing the state" animated="true" width="600"/>
+
+## Listen to the window state
+
+If you need to react to the state changes and send a value to another non-composable application level, for example, write it to the database, you can use `snapshotFlow`.
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:418]:
```suggestion
If you need to react to the state changes and send a value to another non-composable application level, for example, write it to the database, you can use the `snapshotFlow()` function. 
```

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 4]
```diff
@@ -0,0 +1,705 @@
...
+}
+```
+{initial-collapse-state="collapsed" collapsed-title="val state = rememberWindowState(placement = WindowPlacement.Maximized)"}
+
+<img src="compose-window-minimize.png" alt="Changing the state" animated="true" width="600"/>
+
+## Listen to the window state
+
+If you need to react to the state changes and send a value to another non-composable application level, for example, write it to the database, you can use `snapshotFlow`. 
+The `snapshotFlow` function captures the current value of a composable's state.
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:419]:
```suggestion
This function captures the current value of a composable's state.
```

If the previous suggestion is applied, so as not to repeat the name twice.

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 5]
```diff
@@ -0,0 +1,705 @@
...
+import androidx.compose.ui.window.rememberWindowState
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+fun main() = application {
+    val state = rememberWindowState()
+
+    Window(onCloseRequest = ::exitApplication, state) {
+        // Content
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:437]: Not sure what this comment is about :)

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 6]
```diff
@@ -0,0 +1,705 @@
...
+    println("onWindowRelocate $position")
+}
+```
+{initial-collapse-state="collapsed" collapsed-title="LaunchedEffect(state) { snapshotFlow { state.size } .onEach(::onWindowResize)"}
+
+## Dialogs
+
+You can use the `Window` composable to create a regular window and the `DialogWindow` composable for a modal window that locks its parent until the user closes the modal window.
+
+The following code sample demonstrates how to use these composables to combine both regular and modal windows:
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:466]:
```suggestion
The following code sample demonstrates how to use these composables to combine regular and modal windows:
```

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 7]
```diff
@@ -0,0 +1,705 @@
...
+        Button(onClick = { isDialogOpen = true }) {
+            Text(text = "Open dialog")
+        }
+
+        if (isDialogOpen) {
+            DialogWindow(
+                onCloseRequest = { isDialogOpen = false },
+                state = rememberDialogState(position = WindowPosition(Alignment.Center))
+            ) {
+                // Content
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:497]: I can _guess_ what this comment is about, but it could be more transparent.

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 8]
```diff
@@ -0,0 +1,705 @@
...
+            }
+        }
+    }
+}
+```
+{initial-collapse-state="collapsed" collapsed-title="if (isDialogOpen) { DialogWindow(onCloseRequest = { isDialogOpen = false },"}
+
+## Draggable window area
+
+If you want to add a custom draggable title bar to the undecorated window or make the whole window draggable, you can use the `WindowDraggableArea` composable:
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:507]:
```suggestion
To add a custom draggable title bar to the undecorated window or make the whole window draggable, you can use the `WindowDraggableArea` composable:
```

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 9]
```diff
@@ -0,0 +1,705 @@
...
+    Window(onCloseRequest = ::exitApplication, undecorated = true) {
+        WindowDraggableArea {
+            Box(Modifier.fillMaxWidth().height(48.dp).background(Color.DarkGray))
+        }
+    }
+}
+```
+{initial-collapse-state="collapsed" collapsed-title="WindowDraggableArea { Box(Modifier.fillMaxWidth().height(48.dp).background(Color.DarkGray))}"}
+
+`WindowDraggableArea` can be used inside `singleWindowApplication`, `Window`, and `DialogWindow` only. To use it in another composable function, pass `WindowScope` as a receiver:
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:531]:
```suggestion
`WindowDraggableArea` can be used inside `singleWindowApplication()`, `Window()`, and `DialogWindow()` composables only. To use it in another composable function, pass a `WindowScope()` function as a receiver:
```

Unless it is clearly detrimental, we use regular nouns so that the grammar doesn't depend on the user knowing the objects behind the names by heart or figuring it out in context.

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 10]
```diff
@@ -0,0 +1,705 @@
...
+import androidx.compose.runtime.*
+
+fun main() = application {
+    var isOpen by remember { mutableStateOf(true) }
+    if (isOpen) {
+        Window(
+            onCloseRequest = { isOpen = false },
+            title = "Transparent Window Example",
+            transparent = true,
+            undecorated = true, // Transparent window must be undecorated
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:592]: Let's move the comments to a separate line in this sample, too.

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 11]
```diff
@@ -0,0 +1,705 @@
...
+        setContent {
+            // Content
+        }
+        isVisible = true
+    }
+}
+```
+{initial-collapse-state="collapsed" collapsed-title="SwingUtilities.invokeLater { ComposeWindow().apply {"}
+
+You can also access `ComposeWindow` in the composable `Window` scope:
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:630]: As I don't see these names in the code sample, it is unclear how it illustrates this statement. Maybe we should add some context here?

File: topics/compose/compose-desktop-top-level-windows-management.md [hunk 12]
```diff
@@ -0,0 +1,705 @@
...
+        }
+    },
+    dispose = FileDialog::dispose
+)
+```
+{initial-collapse-state="collapsed" collapsed-title="@Composable private fun FileDialog( parent: Frame? = null, "}
+
+## What's next
+
+Explore the tutorials about [other desktop components](https://github.com/JetBrains/compose-multiplatform/tree/master/tutorials#desktop).
```
Comment (zamulla) [inline topics/compose/compose-desktop-top-level-windows-management.md:705]: Should we continue to link GitHub here? I think all of the tutorials are listed in the KMP docs now?

**Review by zamulla — CHANGES_REQUESTED** [review]
Huge work! Some suggestions and questions in the comments.

File: topics/compose/compose-desktop-top-level-windows-management.md (hunk 5 above)
Comment (ElviraMustafina) [reply topics/compose/compose-desktop-top-level-windows-management.md]: Changed to "Content of the window" everywhere

File: topics/compose/compose-desktop-top-level-windows-management.md (hunk 12 above)
Comment (ElviraMustafina) [reply topics/compose/compose-desktop-top-level-windows-management.md:706]: Not yet!

**Review by zamulla — APPROVED** [review]
👍 

There are still build problems with the anchors (because of the changed headers), don't forget to fix them before merging :)
